src/components/SingleTapButton.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { GestureResponderEvent } from "react-native";
+import debounce from "just-debounce-it";
+import Button from "./Button";
+import { ButtonProps } from "./typings/Button";
+
+export interface SingleTapButtonState {
+  loading: boolean;
+}
+
+export interface SingleTapButtonProps extends Omit<ButtonProps, "onPress"> {
+  onPress: (e: GestureResponderEvent) => Promise<any>;
+}
+
+export default class SingleTapButton extends React.PureComponent<
+  SingleTapButtonProps,
+  SingleTapButtonState
+> {
+  state: Readonly<SingleTapButtonState> = {
+    loading: false
+  };
+  private onPress: ButtonProps["onPress"] = debounce(
+    (e: GestureResponderEvent) => {
+      this.setState({ loading: true });
+      // Using function form of finally as want to error out if not returned a Promise
+      this.props.onPress(e).finally(() => {
+        this.setState({ loading: false });
+      });
+    },
+    2000,
```

Comment:
should be set from prop with default value.

Reply:
fixed in e41152e